### ISSUE TITLE
[BUG] Plugin BetterTeams v4.6.4 generated an exception while executing task 13037 java.lang.IllegalStateException: Unregistered scoreboard component

### ISSUE THREAD

**Issue by rishabhk571** [booksaw/BetterTeams#437]
### What is the bug?

[14:53:31 WARN]: [BetterTeams] Plugin BetterTeams v4.6.4 generated an exception while executing task 13037
java.lang.IllegalStateException: Unregistered scoreboard component
        at org.bukkit.craftbukkit.v1_19_R2.scoreboard.CraftTeam.checkState(CraftTeam.java:403) ~[paper-1.19.3.jar:git-Paper-448]
        at org.bukkit.craftbukkit.v1_19_R2.scoreboard.CraftTeam.addEntry(CraftTeam.java:232) ~[paper-1.19.3.jar:git-Paper-448]
        at com.booksaw.betterTeams.events.MCTeamManagement.displayBelowName(MCTeamManagement.java:62) ~[BetterTeams.jar:?]
        at com.booksaw.betterTeams.events.MCTeamManagement.lambda$playerJoinEvent$0(MCTeamManagement.java:120) ~[BetterTeams.jar:?]
        at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftTask.run(CraftTask.java:101) ~[paper-1.19.3.jar:git-Paper-448]
        at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:57) ~[paper-1.19.3.jar:git-Paper-448]
        at com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22) ~[paper-1.19.3.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]

### Expected Behaviour

Nametag should be visible and also in TAB the tag of team

### How to Reproduce

/team tag (tag name)
it shows in tab and nametag but the tag dissapears when we move and then doesn't come back

### Error message

[14:53:31 WARN]: [BetterTeams] Plugin BetterTeams v4.6.4 generated an exception while executing task 13037
java.lang.IllegalStateException: Unregistered scoreboard component
        at org.bukkit.craftbukkit.v1_19_R2.scoreboard.CraftTeam.checkState(CraftTeam.java:403) ~[paper-1.19.3.jar:git-Paper-448]
        at org.bukkit.craftbukkit.v1_19_R2.scoreboard.CraftTeam.addEntry(CraftTeam.java:232) ~[paper-1.19.3.jar:git-Paper-448]
        at com.booksaw.betterTeams.events.MCTeamManagement.displayBelowName(MCTeamManagement.java:62) ~[BetterTeams.jar:?]
        at com.booksaw.betterTeams.events.MCTeamManagement.lambda$playerJoinEvent$0(MCTeamManagement.java:120) ~[BetterTeams.jar:?]
        at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftTask.run(CraftTask.java:101) ~[paper-1.19.3.jar:git-Paper-448]
        at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:57) ~[paper-1.19.3.jar:git-Paper-448]
        at com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22) ~[paper-1.19.3.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]

### Output of /teama version

[15:10:47 INFO]: [Helvetica] Storage Method: com.booksaw.betterTeams.team.storage.storageManager.SeparatedYamlStorageManager
[15:10:47 INFO]: [Helvetica] Minecraft Version: git-Paper-448 (MC: 1.19.3)
[15:10:47 INFO]: [Helvetica] Language Selected: messages
[15:10:47 INFO]: [Helvetica] Current plugin version: 4.6.4

**Comment by rishabhk571** [comment]
[16:18:56 WARN]: [BetterTeams] Plugin BetterTeams v4.6.4 generated an exception while executing task 9485
java.lang.IllegalStateException: Unregistered scoreboard component
        at org.bukkit.craftbukkit.v1_19_R2.scoreboard.CraftTeam.checkState(CraftTeam.java:403) ~[paper-1.19.3.jar:git-Paper-448]
        at org.bukkit.craftbukkit.v1_19_R2.scoreboard.CraftTeam.addEntry(CraftTeam.java:232) ~[paper-1.19.3.jar:git-Paper-448]
        at com.booksaw.betterTeams.events.MCTeamManagement.displayBelowName(MCTeamManagement.java:62) ~[BetterTeams.jar:?]
        at com.booksaw.betterTeams.events.MCTeamManagement.lambda$playerJoinEvent$0(MCTeamManagement.java:120) ~[BetterTeams.jar:?]
        at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftTask.run(CraftTask.java:101) ~[paper-1.19.3.jar:git-Paper-448]
        at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:57) ~[paper-1.19.3.jar:git-Paper-448]
        at com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22) ~[paper-1.19.3.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]

**Comment by rishabhk571** [comment]
[12:10:02 WARN]: [BetterTeams] Plugin BetterTeams v4.6.4 generated an exception while executing task 27442
java.lang.IllegalStateException: Unregistered scoreboard component
        at org.bukkit.craftbukkit.v1_19_R2.scoreboard.CraftTeam.checkState(CraftTeam.java:403) ~[paper-1.19.3.jar:git-Paper-448]
        at org.bukkit.craftbukkit.v1_19_R2.scoreboard.CraftTeam.addEntry(CraftTeam.java:232) ~[paper-1.19.3.jar:git-Paper-448]
        at com.booksaw.betterTeams.events.MCTeamManagement.displayBelowName(MCTeamManagement.java:62) ~[BetterTeams.jar:?]
        at com.booksaw.betterTeams.events.MCTeamManagement.lambda$playerJoinEvent$0(MCTeamManagement.java:120) ~[BetterTeams.jar:?]
        at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftTask.run(CraftTask.java:101) ~[paper-1.19.3.jar:git-Paper-448]
        at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:57) ~[paper-1.19.3.jar:git-Paper-448]
        at com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22) ~[paper-1.19.3.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]

**Comment by rishabhk571** [comment]
[12:15:47 INFO]: Server Plugins (17):
[12:15:47 INFO]: Bukkit Plugins:
[12:15:47 INFO]:  - AuthMe, BanFromClaim, BetterTeams, CoreProtect, DiscordSRV, floodgate, Geyser-Spigot, GriefPrevention, ItemSwapperPlugin, LuckPerms
[12:15:47 INFO]:  PlaceholderAPI, SimpleScore, *SimpleTpa, *UltimateAntiBot, UltimateHomes, ViaBackwards, ViaVersion

**Comment by rishabhk571** [comment]
[05:50:09 WARN]: [BetterTeams] Plugin BetterTeams v4.6.4 generated an exception while executing task 14040
java.lang.IllegalStateException: Unregistered scoreboard component
        at org.bukkit.craftbukkit.v1_19_R2.scoreboard.CraftTeam.checkState(CraftTeam.java:403) ~[paper-1.19.3.jar:git-Paper-448]
        at org.bukkit.craftbukkit.v1_19_R2.scoreboard.CraftTeam.addEntry(CraftTeam.java:232) ~[paper-1.19.3.jar:git-Paper-448]
        at com.booksaw.betterTeams.events.MCTeamManagement.displayBelowName(MCTeamManagement.java:62) ~[BetterTeams.jar:?]
        at com.booksaw.betterTeams.events.MCTeamManagement.lambda$playerJoinEvent$0(MCTeamManagement.java:120) ~[BetterTeams.jar:?]
        at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftTask.run(CraftTask.java:101) ~[paper-1.19.3.jar:git-Paper-448]
        at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:57) ~[paper-1.19.3.jar:git-Paper-448]
        at com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22) ~[paper-1.19.3.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]

**Comment by rishabhk571** [comment]
```
[06:02:24 ERROR]: Could not pass event PlayerQuitEvent to BetterTeams v4.6.4
java.lang.IllegalStateException: Unregistered scoreboard component
        at org.bukkit.craftbukkit.v1_19_R2.scoreboard.CraftTeam.checkState(CraftTeam.java:403) ~[paper-1.19.3.jar:git-Paper-448]
        at org.bukkit.craftbukkit.v1_19_R2.scoreboard.CraftTeam.unregister(CraftTeam.java:310) ~[paper-1.19.3.jar:git-Paper-448]
        at com.booksaw.betterTeams.team.storage.storageManager.SeparatedYamlStorageManager.unloadTeam(SeparatedYamlStorageManager.java:138) ~[BetterTeams.jar:?]
        at com.booksaw.betterTeams.team.storage.storageManager.SeparatedYamlStorageManager.onLeave(SeparatedYamlStorageManager.java:424) ~[BetterTeams.jar:?]
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor155.execute(Unknown Source) ~[?:?]
        at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:77) ~[paper-api-1.19.3-R0.1-SNAPSHOT.jar:?]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[paper-api-1.19.3-R0.1-SNAPSHOT.jar:git-Paper-448]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[paper-api-1.19.3-R0.1-SNAPSHOT.jar:?]
        at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54) ~[paper-1.19.3.jar:git-Paper-448]
        at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126) ~[paper-1.19.3.jar:git-Paper-448]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:615) ~[paper-api-1.19.3-R0.1-SNAPSHOT.jar:?]
        at net.minecraft.server.players.PlayerList.remove(PlayerList.java:578) ~[paper-1.19.3.jar:git-Paper-448]
        at net.minecraft.server.players.PlayerList.remove(PlayerList.java:563) ~[paper-1.19.3.jar:git-Paper-448]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.onDisconnect(ServerGamePacketListenerImpl.java:2119) ~[?:?]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.onDisconnect(ServerGamePacketListenerImpl.java:2098) ~[?:?]
        at net.minecraft.network.Connection.handleDisconnection(Connection.java:777) ~[?:?]
        at net.minecraft.server.network.ServerConnectionListener.tick(ServerConnectionListener.java:253) ~[?:?]
        at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1565) ~[paper-1.19.3.jar:git-Paper-448]
        at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:447) ~[paper-1.19.3.jar:git-Paper-448]
        at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1397) ~[paper-1.19.3.jar:git-Paper-448]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1173) ~[paper-1.19.3.jar:git-Paper-448]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316) ~[paper-1.19.3.jar:git-Paper-448]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

**Comment by rishabhk571** [comment]
```
[06:07:59 INFO]: gamespy_123 issued server command: /team disband
[06:07:59 ERROR]: Something went wrong while executing the command, please report this https://github.com/booksaw/BetterTeams/issues/new/choose
[06:07:59 WARN]: java.lang.IllegalStateException: Unregistered scoreboard component
[06:07:59 WARN]:        at org.bukkit.craftbukkit.v1_19_R2.scoreboard.CraftTeam.checkState(CraftTeam.java:403)
[06:07:59 WARN]:        at org.bukkit.craftbukkit.v1_19_R2.scoreboard.CraftTeam.unregister(CraftTeam.java:310)
[06:07:59 WARN]:        at BetterTeams.jar//com.booksaw.betterTeams.Team.disband(Team.java:742)
[06:07:59 WARN]:        at BetterTeams.jar//com.booksaw.betterTeams.commands.team.DisbandCommand.onCommand(DisbandCommand.java:49)
[06:07:59 WARN]:        at BetterTeams.jar//com.booksaw.betterTeams.commands.presets.TeamSubCommand.onCommand(TeamSubCommand.java:39)
[06:07:59 WARN]:        at BetterTeams.jar//com.booksaw.betterTeams.commands.ParentCommand.onCommand(ParentCommand.java:135)
[06:07:59 WARN]:        at BetterTeams.jar//com.booksaw.betterTeams.BooksawCommand.runExecution(BooksawCommand.java:72)
[06:07:59 WARN]:        at BetterTeams.jar//com.booksaw.betterTeams.BooksawCommand.lambda$execute$0(BooksawCommand.java:59)
[06:07:59 WARN]:        at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftTask.run(CraftTask.java:101)
[06:07:59 WARN]:        at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:57)
[06:07:59 WARN]:        at com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22)
[06:07:59 WARN]:        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[06:07:59 WARN]:        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[06:07:59 WARN]:        at java.base/java.lang.Thread.run(Thread.java:833)
```

**Comment by rishabhk571** [comment]
Solved, just needed to update to 1.19.3 dev version of Protocol Lib

**Comment by booksaw** [comment]
@rishabhk571 Thanks for providing information on how to solve, that makes my life a lot easier
